### ISSUE TITLE
FLERT predictions on document level

### ISSUE THREAD

**Issue by agademic** [flairNLP/flair#2350]
I've trained a model in the fashion of FLERT and I am pretty happy with the accuracy on the test set. Now I would like to predict another unseen dataset (documents split in sentences) but I am not sure how to trigger the tagger to consider the document as a whole.

Any hints are highly appreciated!

**Comment by alanakbik** [comment]
It depends on how you load the dataset and call the tagger. If you pass a list of sentences to the `.predict()` method, the tagger will automatically consider all sentences within the same mini-batch to belong to the same document. So you essentially have document context automatically. If you want to make sure that no document boundaries are crossed, you could pass each document as a list of sentences to the predict method individually.

**Comment by agademic** [comment]
Hmm, that's interesting. I was passing the input in the very same fashion you recommend.
While passing the document as a list of sentences, the predictions were quite bad. Then I tried to pass the document as a single string (with `use_tokenizer=True`) in the `Sentence` object and I got great results! I am wondering how the max document length will affect results here.

**Comment by alanakbik** [comment]
Really? Hm that should not happen. Could you share a code snippet how you call the tagger? Are you on master branch or latest release?

**Comment by agademic** [comment]
This was my first approach (same as you recommended):
```
sentences = []
doc = nlp(document) # spacy transformer lm here, to split document in sentences, training data was split the same way
for sent in doc.sents:
    sentences.append(Sentence(str(sent), use_tokenizer=True))
model.predict(sentences)

for sentence in sentences:
    sentence.to_dict(tag_type='ner')
```
That approach led to disappointing results.

The current call simply is:
```
sent = Sentence(document, use_tokenizer=True)
model.predict(sent)
sent.to_dict(tag_type='ner)
```
Predictions here are perfectly fine (approx. same F1 as in test data).

I am using the latest release (0.8.0.post1).

**Comment by alanakbik** [comment]
I can't reproduce this behavior. I'm trying with a FLERT model and two sentence one by one (case 1) and two sentences as one document (case 2), but I get the same results. See code below. Could you adapt this code to your model and check if you see the same behavior?

```python
from flair.data import Sentence
from flair.models import SequenceTagger

# load FLERT tagger
model = SequenceTagger.load("ner-large")

# example document with two sentences
doc_sents = ["This is Washington.", "I mean George Washington."]

#### CASE 1: document as two separate sentences #### 
sentences = []
for sent in doc_sents:
    sentences.append(Sentence(str(sent), use_tokenizer=True))

# predict
model.predict(sentences)

# print predictions
for sentence in sentences:
    print(sentence.to_original_text())
    for entity in sentence.get_spans('ner'):
        print(f"{entity}")


#### CASE 2: make a single sentence for document #### 
sentence = Sentence(" ".join(doc_sents))

# predict
model.predict(sentence)

# print predictions
print(sentence.to_original_text())
for entity in sentence.get_spans('ner'):
    print(f"{entity}")
```

**Comment by agademic** [comment]
First of all:
Thank you so much for your energy and your time looking into my question!

I adapted your code and you are completely right. The results are identical in both cases. I found that in my previous experiments I was iterating over the sentences, calling `Sentence` and `.predict()` on each sentence separately.

Good thing is though: realizing that the `predict` method can handle single strings without the pain of using a sentencizer.
Another finding: the HuggingFace error `Token indices sequence length is longer than the specified maximum sequence length for this model (889 > 512). Running this sequence through the model will result in indexing errors` is popping up in the single string case (obviously) but it doesn't if we call `predict` on separate sentences.
